packages/flutter_tools/lib/src/base/dds.dart

```diff
@@ -55,15 +50,15 @@ class DartDevelopmentService {
       _ddsInstance = await ddsLauncherCallback(
           observatoryUri,
           serviceUri: ddsUri,
-          enableAuthCodes: !disableServiceAuthCodes,
-          ipv6: ipv6,
+          enableAuthCodes: disableServiceAuthCodes == true,
+          ipv6: ipv6 == true,
         );
-      unawaited(_ddsInstance.done.whenComplete(() {
+      unawaited(_ddsInstance?.done.whenComplete(() {
```

Comment:
I'm confused by `_ddsInstance?.done.whenComplete(())`. Wouldn't we get a NPE if `_ddsInstance == null`? (I'm assuming I'm wrong and you and the analyzer are correct though...)

Reply:
If it's changed to `_ddsInstance?.done?.whenComplete` you get:
```
warning: The receiver can't be null because of short-circuiting, so the null-aware operator '?.' can't be used. (invalid_null_aware_operator at [flutter_tools] lib/src/base/dds.dart:56)
```
`_ddsInstance` is nullable, but once it's established to be nonnull, `done` will also be nonnull.
```
Future<void> get done;
```

Reply:
ahh, ok, i didn't know `?.` did short circuiting.